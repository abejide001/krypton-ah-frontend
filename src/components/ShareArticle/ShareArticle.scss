.SocialMediaShareButton{
  display: grid;
  margin: 1.3rem;
  outline: none;
  cursor: pointer;
}
