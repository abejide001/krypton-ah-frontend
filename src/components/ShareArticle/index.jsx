import React from 'react';
import PropTypes from 'prop-types';
import {
  FacebookShareButton,
  GooglePlusShareButton,
  LinkedinShareButton,
  TwitterShareButton,
  WhatsappShareButton,
  EmailShareButton,
  FacebookIcon,
  TwitterIcon,
  GooglePlusIcon,
  LinkedinIcon,
  WhatsappIcon,
  EmailIcon } from 'react-share';
import { API_BASE_URL } from '../../constants';
import './ShareArticle.scss';

class ShareArticle extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const { title, slug, description } = this.props;
    const url = `${API_BASE_URL}/${slug}`;
    const hashtags = ['ah-Krypton', 'Andela', 'Readers'];
    return (
      <div className="shareArticleIcons">
        <FacebookShareButton quote={description} url={url}>
          <FacebookIcon size="25" round />
        </FacebookShareButton>
        <GooglePlusShareButton quote={description} url={url}>
          <GooglePlusIcon size={25} round />
        </GooglePlusShareButton>
        <TwitterShareButton title={title} via="Andela" hashtags={hashtags} url={url}>
          <TwitterIcon size={25} round />
        </TwitterShareButton>
        <LinkedinShareButton title={title} description={description} url={url}>
          <LinkedinIcon size={25} round />
        </LinkedinShareButton>
        <WhatsappShareButton title={title} url={url}>
          <WhatsappIcon size={25} round />
        </WhatsappShareButton>
        <EmailShareButton subject={title} body={description} url={url}>
          <EmailIcon size={25} round />
        </EmailShareButton>

      </div>
    );
  }
}

ShareArticle.propTypes = {
  title: PropTypes.string.isRequired,
  slug: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

export default ShareArticle;
