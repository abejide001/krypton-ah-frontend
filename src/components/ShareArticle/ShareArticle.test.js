import React from 'react';
import { shallow } from 'enzyme';
import ShareArticle from '.';

describe('Share Article Component', () => {
  it('should rendered share Article Icons without crashing', () => {
    const wrapper = shallow(<ShareArticle
    title="news" />);
    expect(wrapper).toMatchSnapshot();
  });
});
