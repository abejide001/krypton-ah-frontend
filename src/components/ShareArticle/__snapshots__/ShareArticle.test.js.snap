// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Share Article Component should rendered share Article Icons without crashing 1`] = `
<div
  className="shareArticleIcons"
>
  <CreatedButton
    quote=" "
    url="undefined/article/ "
    windowHeight={400}
    windowWidth={550}
  >
    <Icon
      className=""
      iconBgStyle={Object {}}
      logoFillColor="white"
      round={true}
      size={25}
    />
  </CreatedButton>
  <CreatedButton
    hashtags={
      Array [
        "ah-Krypton",
      ]
    }
    title=" "
    url="undefined/article/ "
    windowHeight={400}
    windowWidth={550}
  >
    <Icon
      className=""
      iconBgStyle={Object {}}
      logoFillColor="white"
      round={true}
      size={25}
    />
  </CreatedButton>
  <CreatedButton
    description=" "
    title=" "
    url="undefined/article/ "
    windowHeight={600}
    windowWidth={750}
  >
    <Icon
      className=""
      iconBgStyle={Object {}}
      logoFillColor="white"
      round={true}
      size={25}
    />
  </CreatedButton>
  <CreatedButton
    separator=" "
    title=" "
    url="undefined/article/ "
    windowHeight={400}
    windowWidth={550}
  >
    <Icon
      className=""
      iconBgStyle={Object {}}
      logoFillColor="white"
      round={true}
      size={25}
    />
  </CreatedButton>
  <CreatedButton
    body=" (<br />) https://krypton-ah-fe-stage.herokuapp.com/article/ "
    onClick={[Function]}
    openWindow={false}
    subject=" "
    url="undefined/article/ "
  >
    <Icon
      className=""
      iconBgStyle={Object {}}
      logoFillColor="white"
      round={true}
      size={25}
    />
  </CreatedButton>
</div>
`;
